### PR TITLE
Add Swift Global Replication Tests

### PR DESCRIPTION
**Pull request by gnuoy**

### CODE REVIEW

File: zaza/openstack/utilities/swift.py [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+        return cmd_result['Stdout']
+
+    @property
+    def hand_off_ips(self):
+        """Replicas which are marked as handoff devices.
+
+        These are not real replicas. They hand off the replica to other node.
+
+        :returns: List of IPS of handoff nodes for object.
+        :rtype: [str, ...]
```
Comment (ajkavanagh) [inline zaza/openstack/utilities/swift.py]: In official types, this is `List[str]`

File: zaza/openstack/utilities/swift.py [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+        """
+        return [r.server for r in self.replicas if not r.handoff_device]
+
+    @property
+    def placements(self):
+        """Region an zone information for each replica.
+
+        :returns: List of dicts with region and zone information.
+        :rtype: [{
```
Comment (ajkavanagh) [inline zaza/openstack/utilities/swift.py]: In official Python types this is: `List[Dict[str, Union[str,int]]]`

File: zaza/openstack/utilities/swift.py [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+        :rtype: [int, ...]
+        """
+        return list(set([p['region'] for p in self.placements]))
+
+    @property
+    def all_zones(self):
+        """List of all zones that have a replica.
+
+        :returns: List of tuples (region, zone) that have a replica.
+        :rtype: [(r1, z1), ...]
```
Comment (ajkavanagh) [inline zaza/openstack/utilities/swift.py]: In python types, this is `List[Tuple[str, str]`

File: zaza/openstack/utilities/swift.py [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+
+
+def setup_test_container(swift_client, resource_prefix):
+    """Create a swift container for use be tests.
+
+    :param swift_client: Swift client to use for object creation
+    :type swift_client: swiftclient.Client
+    :returns: (container_name, account_name) Container name and account
+              name for new container
+    :rtype: (str, str)
```
Comment (ajkavanagh) [inline zaza/openstack/utilities/swift.py]: Officially, python type is `Tuple[str, str]`

File: unit_tests/utilities/swift_test_data.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
+# flake8: noqa
+
+SWIFT_GET_NODES_STDOUT = """
```
Comment (ajkavanagh) [inline unit_tests/utilities/swift_test_data.py]: Does this command really return all the `curl` commands in the output too?

File: zaza/openstack/charm_tests/swift/tests.py [hunk 6]
```diff
@@ -110,3 +113,115 @@ def test_901_pause_resume(self):
...
+            model_name=cls.region1_model_name)
+        cls.swift_region1 = openstack_utils.get_swift_session_client(
+            cls.swift_session,
+            region_name='RegionOne')
+        cls.swift_region2 = openstack_utils.get_swift_session_client(
+            cls.swift_session,
+            region_name='RegionTwo')
+
+    @classmethod
+    @tenacity.retry(
```
[the remark this answers is not in the document]
Comment (gnuoy) [reply zaza/openstack/charm_tests/swift/tests.py]: The method does delete the objects before the container. However the tenacity decorator is really there to work around the issue that swift container deletion seems random at best. I have updated the doc string to reflect that.

**Pull request merged**